src/main/java/com/coreos/jetcd/ClientBuilder.java

```diff
@@ -14,13 +14,17 @@
 import io.grpc.NameResolver;
 import io.grpc.netty.GrpcSslContexts;
 import io.netty.handler.ssl.SslContext;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * ClientBuilder knows how to create an Client instance.
  */
 public class ClientBuilder implements Cloneable {
-
+  private static final List<String> SCHEMES = Arrays.asList("http", "https", "unix", "unixs");
```

Comment:
Is unix supported by default ?  As far as I can remember the SimpleNameResolver explicit checks for http/https. In addition the endpoint format depends on the NameResolver which is being used so maybe the check should be moved on the *NameResolver* class.